mix.exs

```diff
@@ -85,20 +81,29 @@ defmodule CouchDBTest.Mixfile do
 
   # Run "mix help deps" to learn about dependencies.
   defp deps() do
-    deps_list = [
+    deps1 = [
       {:junit_formatter, "~> 3.0", only: [:dev, :test, :integration]},
       {:httpotion, ">= 3.1.3", only: [:dev, :test, :integration], runtime: false},
       {:excoveralls, "~> 0.12", only: :test},
-      {:b64url, path: path("b64url")},
-      {:jiffy, path: path("jiffy")},
-      {:jwtf, path: path("jwtf")},
       {:ibrowse, path: path("ibrowse"), override: true},
       {:credo, "~> 1.7.7", only: [:dev, :test, :integration], runtime: false}
     ]
 
+    extra_deps = [:b64url, :jiffy, :jwtf, :meck, :mochiweb]
+    deps2 = Enum.map(extra_deps, &{&1, path: path(&1)})
+
+    deps_list =
+      if System.get_env("CI") do
```

Comment:
Wonder why we need this. Ideally a CI run shouldn't look too much different than a local dev run, otherwise it could become tricky to debug things locally (works in CI but not locally, or works locally but fail in CI) so could be result in some frustration down the way.

Reply:
```elixir
File.mkdir_p!(Path.dirname(build_ebin))
```
In Jenkins, [this function](https://github.com/apache/couchdb/pull/5372/files#diff-dfa6f4ed74c90e5d4fda283d547d366586e690387289bcfd473e3fa5f9ace308R120) can't create folders in `_build`, so it can't create a symbolic link to the ebin folder of fabric/config. The [error message](https://ci-couchdb.apache.org/job/jenkins-cm1/job/PullRequests/job/PR-5372/9/execution/node/88/log/) is:
`13:19:19  /usr/bin/ln: failed to create symbolic link '/home/jenkins/workspace/jenkins-cm1_PullRequests_PR-5372/_build/dev/lib/config/ebin': No such file or directory`

Reply:
> works in CI but not locally, or works locally but fail in CI

Some warnings like the following are removed locally but remained in CI.
```
    warning: :fabric.create_db/2 is undefined (module :fabric is not available or is yet to be defined)
    │
 39 │     res = :fabric.create_db(name, [@admin])
    │                   ~
    │
    └─ test/elixir/lib/step/create_db.ex:39:19: Couch.Test.Setup.Step.Create.DB.setup/2
```

Reply:
A general question, why do we need this linking to the ebin dir?

Reply:
fabric and config are defined in the `src` folder, but when compiling the code, `mix` cannot find them in the `_build` folder, that's why we saw these warnings.

Reply:
It seems `code.add_patha/1` works locally and CI, thank you for the review!